### ISSUE TITLE
Support dropdown of generic names and for property mediator `name` and field

### ISSUE THREAD

**Issue by poorna2152** [wso2/mi-vscode#153]
**Description:**
Based on the description of the `Name` property of the property mediator given [here](https://mi.docs.wso2.com/en/4.3.0/reference/mediators/property-mediator/). The `Property Name` property could be a expression and it should also give a dropdown of possible generic names in that field.
 
But in the current form of the property mediator it is not possible to mark the name as a expression and there is no dropdown of generic names for the name field of the property.

<img width="539" alt="Screenshot 2024-07-25 at 09 27 24" src="https://github.com/user-attachments/assets/fbf4bf19-ca52-4e66-8d3f-c45e20788346">

**Comment by thuva9872** [comment]
The expression support for property name is added in the latest release v1.0.2.

**Comment by thuva9872** [comment]
We haven't finalised the dropdown of generic names and haven't planned to give that feature for now. This feature will be added in the future release.

**Comment by ChinthakaJ98** [comment]
To address this issue we need a catalog which could be used to give users the option of either a browsing window or an AI search to select the relevant property name. Therefore this catalog creation needs to be done first in a separate effort in order to provide the required UI support from the MI extension and that effort is being tracked in issue https://github.com/wso2/micro-integrator/issues/3620
